src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -907,14 +915,21 @@ private static LLVMValueRef BuildConstInt32(int number)
             return LLVM.ConstInt(LLVM.Int32Type(), (ulong)number, LLVMMisc.False);
         }
 
-        private LLVMValueRef GetEETypeForTypeDesc(TypeDesc target)
+        private LLVMValueRef GetEETypeForTypeDesc(TypeDesc target, bool constructed)
         {
-            ISymbolNode node = _compilation.NodeFactory.ConstructedTypeSymbol(target);
+            ISymbolNode node;
```

Comment:
Can this just use `_compilation.NodeFactory.MaximallyConstructableType`, it seems like the only difference is that here you're baking in the knowledge of constructed or not, while in the utility function it would be determined by `ConstructedEETypeNode.CreationAllowed`

Reply:
This is a bit more efficient because making a type constructed makes its method bodies necessary. If we only refer to a type for ldtoken or cast or something like that, we don't need it to be constructed.